composer.json

```
@@ -61,14 +61,14 @@
         "illuminate/validation": "^6.0",
         "illuminate/view": "^6.0",
         "intervention/image": "^2.5.0",
-        "laminas/laminas-diactoros": "^1.8.4",
-        "laminas/laminas-httphandlerrunner": "^1.0",
-        "laminas/laminas-stratigility": "^3.0",
+        "laminas/laminas-diactoros": "^2.4.1 || ^2.5.0",
```

Why can't we just use the former of the conditional requirements? Wouldn't the effect be the same, since caret goes all the way up to the next major version, and all of these are post-1.0 versions?

---

Because only those two versions are currently compatible. The old constraint `^1.8.4` would only allow upgrade up to `1.9` and blocks `1.9.0` and up.

The conditional requirement is required because 2.4.1 and up is compatible with php 7.3+; and only 2.5.0 and up is compatible with php 8.

---

I think what he means is, isn't `^2.4.1` no different than `^2.4.1 || ^2.5.0` ?
https://jubianchi.github.io/semver-check/#/^2.4.1/2.5.0

---

😱 I mixed up tilde against caret, sorry folks. Change makes sense!
```suggestion
        "laminas/laminas-diactoros": "^2.4.1",
```